### PR TITLE
feat(cloudsql/postgres): update to V2 sample.

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

**Review by enocom — APPROVED** [review]

File: cloud-sql/postgres/knex/app.standard.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-runtime: nodejs10
+runtime: nodejs17
```
Comment (ace-n) [inline cloud-sql/postgres/knex/app.standard.yaml:14]: Even-numbered Node releases are used as long-term support (LTS) candidates.

Perhaps we should use `node16` here?

File: cloud-sql/postgres/knex/index.js [hunk 2]
```diff
@@ -311,7 +240,7 @@ app.get('/', async (req, res) => {
   }
 });
 
-app.post('/', async (req, res) => {
+const httpPost = app.post('*', async (req, res) => {
```
Comment (ace-n) [inline cloud-sql/postgres/knex/index.js:243]: `httpPost` looks like it points to an Express route, not the route handler function itself.

We probably need something like this:
```
const httpPost = async (req, res) => {
   ...
}

app.post('*', httpPost)
```

**Review by ace-n — APPROVED** [review]
LGTM once:
a) comments addressed, **and**
b) tests pass

File: cloud-sql/postgres/knex/connect-unix.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+// [START cloud_sql_postgres_knex_connect_unix]
+const Knex = require('knex');
+
+const createUnixSocketPool = async config => {
+  const dbSocketPath = process.env.INSTANCE_UNIX_SOCKET;
```
Comment (enocom) [inline cloud-sql/postgres/knex/connect-unix.js:20]: Let's add the secret manager disclaimer in these samples. See https://github.com/GoogleCloudPlatform/golang-samples/blob/main/cloudsql/postgres/database-sql/connect_tcp.go#L42-L45.

**Review by sofisl — APPROVED** [review]

**Pull request merged**